### PR TITLE
Added tests and re-configured vitest (i'm sorry)

### PR DESCRIPTION
**Pull request by thisisofficial**
### Description

 Finished with some innitial component tests for the Data Grid component. As well as
 configured vitest to work properly plus the vitest-ui to have more control about the tests.
 
 The reason I moved the project back to vitest was because it's way easier for me to develop for it,
 also it tests faster than jest, shaving a lot of time off testing.

### Testing Checklist

- [x] Test cases have been added/updated
- [x] Tested locally in browser (at least one other than Chrome)
- [x] All tests pass

### What JIRA Ticket(s) does this PR address?

- [FP-4](https://cd95.atlassian.net/browse/FP-4)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "build": "tsc && vite build",
     "preview": "vite preview",
     "prepare": "husky install",
-    "test": "jest",
+    "test": "vitest",
```
Comment (caarlosdamian) [inline package.json:11]: no need for vitest

File: src/pages/DataGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { GridRowsProp } from '@mui/x-data-grid';
```
Comment (caarlosdamian) [inline src/pages/DataGrid.tsx:1]: why added this component this is not related to test component worked fine

File: src/utils/__snapshots__/tableFuntion.test.tsx.snap [hunk 3]
```diff
@@ -0,0 +1,101 @@
+// Vitest Snapshot v1
```
Comment (caarlosdamian) [inline src/utils/__snapshots__/tableFuntion.test.tsx.snap:1]: snapshot testing is obsolite delete this

File: src/utils/tableFuntion.test.tsx [hunk 4]
```diff
@@ -1,11 +1,46 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-disable import/no-extraneous-dependencies */
+import { describe, expect, it } from 'vitest';
```
Comment (caarlosdamian) [inline src/utils/tableFuntion.test.tsx:3]: this is not need it rtl have this remove vitest

File: src/utils/tableFuntion.test.tsx [hunk 5]
```diff
@@ -1,11 +1,46 @@
...
+import { fireEvent, render, screen } from '@testing-library/react';
 import { numberWithCommas } from './functions';
+import { columns } from './tableColumns';
+import { DataGrid } from '../pages/DataGrid';
 
-describe('isBoolean', () => {
-  it('should return the proper separation of numbers in a string', () => {
+describe('Funtions working', () => {
```
Comment (caarlosdamian) [inline src/utils/tableFuntion.test.tsx:9]: this test were good why are you changin it?

File: src/utils/tableFuntion.test.tsx [hunk 6]
```diff
@@ -1,11 +1,46 @@
...
-  it('should return the proper separation of string numbers as a new string', () => {
+  it('Should return the proper separation of string numbers as a new string', () => {
     expect(numberWithCommas('1000')).toBe('1,000');
   });
 });
+
+// Each should be re-snapshotted if changed.
+describe('All Snapshots Match', () => {
+  it("Column's snapshot match", () => {
+    expect(columns).toMatchSnapshot();
```
Comment (caarlosdamian) [inline src/utils/tableFuntion.test.tsx:22]: snapshot are obsolete remove it

File: vite.config.ts [hunk 7]
```diff
@@ -1,7 +1,15 @@
-import { defineConfig } from 'vite'
-import react from '@vitejs/plugin-react'
+/* eslint-disable import/no-extraneous-dependencies */
+/// <reference types="vitest" />
```
Comment (caarlosdamian) [inline vite.config.ts:2]: remove eslint disable no need to this you need to be good with eslint in other words change it all

File: vite.config.ts [hunk 8]
```diff
@@ -1,7 +1,15 @@
...
+/// <reference types="vite/client" />
+
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
 
 // https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [react()]
-})
```
Comment (caarlosdamian) [inline vite.config.ts:7]: this config is not needed testing config is already set

**Review by caarlosdamian — CHANGES_REQUESTED** [review]

File: src/utils/tableFuntion.test.tsx (hunk 5 above)
Comment (thisisofficial) [reply src/utils/tableFuntion.test.tsx:9]: The descriptions and overall writting of the tests were weird, i just changed them so they looked nicer.

Comment (caarlosdamian) [reply src/utils/tableFuntion.test.tsx:9]: no need to change them use the config that is already config

File: package.json (hunk 1 above)
Comment (caarlosdamian) [reply package.json:11]: We are not goingo to accept any other test configurations please check and work whit the current config

File: package.json [hunk 9]
```diff
@@ -24,6 +24,7 @@
     "@types/jest": "^28.1.6",
     "identity-obj-proxy": "^3.0.0",
     "jest-environment-jsdom": "^28.1.3",
+    "node-fetch": "^3.2.10",
```
Comment (caarlosdamian) [inline package.json:27]: no need for node-fetch remove it

**Review by caarlosdamian — CHANGES_REQUESTED** [review]

**Comment by thisisofficial** [comment]
Needed to rebase and re-do this PR.
